core/network/src/protocol.rs

```diff
@@ -1121,18 +1121,12 @@ impl<B: BlockT, S: NetworkSpecialization<B>, H: ExHashT> Protocol<B, S, H> {
 		};
 
 		match self.sync.on_block_announce(who.clone(), hash, &announce, is_their_best) {
-			sync::OnBlockAnnounce::Request(peer, req) => {
-				self.send_message(peer, GenericMessage::BlockRequest(req));
-				return CustomMessageOutcome::None
-			}
 			sync::OnBlockAnnounce::Nothing => {
 				// try_import is only true when we have all data required to import block
 				// in the BlockAnnounce message. This is only when:
 				// 1) we're on light client;
 				// AND
-				// - EITHER 2.1) announced block is stale;
-				// - OR 2.2) announced block is NEW and we normally only want to download this single block (i.e.
-				//           there are no ascendants of this block scheduled for retrieval)
+				// 2) parent block is already imported
```

Comment:
@svyatonik please check if this is correct

Reply:
I'd change that to smth like "parent block is already imported and not pruned" (given my other comment). Also "try_import is only true" should be changed to "on_block_announce returns OnBlockAnnounce::ImportHeader".